tests/network/localnet/liblocalnet.py

```diff
@@ -106,6 +106,7 @@ def localnet_vm(
 
     vmi_spec = spec.template.spec
     vmi_spec.networks = networks
+    vmi_spec.domain.devices = vmi_spec.domain.devices or Devices()
```

Comment:
Why do you need this?

Reply:
To make pre-commit happy - vmi_spec.domain.devices may be None, but .interfaces was accessed without a check.

> tests/network/localnet/liblocalnet.py:109: error: Item "None" of "Devices | None" has no attribute "interfaces"  [union-attr]